src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -584,9 +596,26 @@ case class MergeIntoCommand(
 
   private def getTargetOutputCols(txn: OptimisticTransaction): Seq[NamedExpression] = {
     txn.metadata.schema.map { col =>
-      target.output.find(attr => conf.resolver(attr.name, col.name)).getOrElse {
+      targetOutputAttributesMap.getOrElse(
+        col.name,
         Alias(Literal(null, col.dataType), col.name)()
-      }
+      )
+    }
+  }
+
+  private def getTargetOutputColsMap(
```

Comment:
why not make it a better function that also takes the name of the column that is being searched for in line 577 and return the expression?

Reply:
stepping back, this function is alarmingly similar in logic as `getTargetOutputCols`. pretty sure you can generate this map inside `buildTargetPlanWithFiles` (the only place that uses this function) by calling `getTargetOutputCols`. basically.. you don't need this additional function.

Reply:
Agree, I just tried to not overcomplicate the code to avoid a new iteration through the result of `getTargetOutputCols` to add column names as keys. Anyway, it should not be so expensive, what do you think?